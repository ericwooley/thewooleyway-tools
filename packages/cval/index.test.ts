import * as validate from './'

describe('composeValidations', () => {
  const emailValidation = validate.composeValidations(
    validate.isEmail,
    validate.minLength(9),
    validate.maxLength(15)
  )
  it('should pass email', () =>
    expect(emailValidation('email', 'bob@bob.com')).toMatchSnapshot('composeValidations - email: valid')
  )
  it('should fail for too long', () =>
    expect(emailValidation('email', 'bob@bob.comoTheDoggo')).toMatchSnapshot('composeValidations - email: invalid: too long')
  )
  it('should fail for too short', () =>
    expect(emailValidation('email', 'a@b.c')).toMatchSnapshot('composeValidations - email: invalid: too short')
  )
  it('should fail for invalid', () =>
    expect(emailValidation('email', 'abcd1234123')).toMatchSnapshot('composeValidations - email: invalid: invalid')
  )
  it('should fail for invalid and too short', () =>
    expect(emailValidation('email', 'a')).toMatchSnapshot('composeValidations - email: invalid: invalid & too short')
  )
})

describe('isEmail', () => {
  it('should validate an email', () =>
  expect(validate.isEmail('email', 'bob@bob.com')).toMatchSnapshot('email: valid'))
  it('should invalidate an invalid email', () =>
    expect(validate.isEmail('email', 'steve')).toMatchSnapshot('email: invalid'))
})

describe('noEmpty', () => {
  it('should validate a non-empty string', () =>
  expect(validate.noEmpty('noEmpty', 'random thing')).toMatchSnapshot('noEmpty: valid'))

  it('should invalidate an empty string', () =>
    expect(validate.noEmpty('string', '')).toMatchSnapshot('noEmpty: invalid'))
})

describe('minLength', () => {
  it('should validate a length 12 string', () =>
  expect(validate.minLength(12)('minLenVal', '123456789012')).toMatchSnapshot('minLength - 12: valid'))

  it('should validate a length 13 string', () =>
  expect(validate.minLength(12)('minLenVal', '1234567890123')).toMatchSnapshot('minLength - 13: valid'))

  it('should validate a length 11 string', () =>
  expect(validate.minLength(12)('minLenVal', '12345678901')).toMatchSnapshot('minLength - 11: invalid'))
})

describe('maxLength', () => {
  it('should validate a length 12 string', () =>
  expect(validate.maxLength(12)('maxLenVal', '123456789012')).toMatchSnapshot('maxLength - 12: valid'))

  it('should validate a length 11 string', () =>
  expect(validate.maxLength(12)('maxLenVal', '123456789011')).toMatchSnapshot('maxLength - 11: valid'))

  it('should invalidate a length 13 string', () =>
  expect(validate.maxLength(12)('maxLenVal', '1234567890123')).toMatchSnapshot('maxLength - 13: invalid'))
})

describe('isNumber', () => {
  it('should validate a number', () =>
    expect(validate.isNumber('number', 12)).toMatchSnapshot('number: valid'))

  it('should invalidate an invalid Number', () =>
    expect(validate.isNumber('number', 'steve' as any)).toMatchSnapshot('number: invalid'))
})

describe('mustMatchRegex', () => {
  it('should validate passing regexes',
    () => expect(
      validate.mustMatchRegex(/^[abcd]$/, 'must be a, b, c, or d')('alphabet', 'a')
    ).toMatchSnapshot('mustMatchRegex: valid')
  )
  it('should invalidate passing regexes',
    () => expect(
      validate.mustMatchRegex(/^[abcd]$/, 'must be a, b, c, or d')('alphabet', 'f')
    ).toMatchSnapshot('mustMatchRegex: invalid')
  )
})

describe('mustBeOfType', () => {
  it('should validate passing regexes',
    () => expect(
      validate.mustBeOfType('string')('mustBeofType', 'asdfasdf')
    ).toMatchSnapshot('mustBeOfType: valid')
  )
  it('should invalidate passing regexes',
    () => expect(
      validate.mustBeOfType('string')('mustBeofType', 12)
    ).toMatchSnapshot('mustBeOfType: invalid')
  )
})

describe('lessThan', () => {
  it('should validate a length 12 string', () =>
  expect(validate.lessThan(12)('number', 10)).toMatchSnapshot('lessThan - 12: valid'))

  it('should validate a length 13 string', () =>
  expect(validate.lessThan(12)('number', -13)).toMatchSnapshot('lessThan - -13: valid'))

  it('should validate a length 11 string', () =>
  expect(validate.lessThan(12)('number', 47)).toMatchSnapshot('lessThan - 47: invalid'))
})

describe('moreThan', () => {
  it('should validate a length 25 string', () =>
  expect(validate.moreThan(-12)('number', 25)).toMatchSnapshot('moreThan - 25: valid'))

  it('should validate a length 13 string', () =>
  expect(validate.moreThan(-12)('number', -5)).toMatchSnapshot('moreThan - -5: valid'))

  it('should validate a length 11 string', () =>
  expect(validate.moreThan(12)('number', 0)).toMatchSnapshot('moreThan - 0: invalid'))
})
