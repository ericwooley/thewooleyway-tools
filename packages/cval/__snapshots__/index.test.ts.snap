// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`composeValidations - email: invalid: invalid & too short 1`] = `
Array [
  "email must be a valid email",
  "email must be at least 9 characters",
]
`;

exports[`composeValidations - email: invalid: invalid 1`] = `
Array [
  "email must be a valid email",
]
`;

exports[`composeValidations - email: invalid: too long 1`] = `
Array [
  "email must be at most 15 characters",
]
`;

exports[`composeValidations - email: invalid: too short 1`] = `
Array [
  "email must be at least 9 characters",
]
`;

exports[`composeValidations - email: valid 1`] = `Array []`;

exports[`email: invalid 1`] = `"email must be a valid email"`;

exports[`email: valid 1`] = `""`;

exports[`lessThan - -13: valid 1`] = `""`;

exports[`lessThan - 12: valid 1`] = `""`;

exports[`lessThan - 47: invalid 1`] = `"number must be at less than 12"`;

exports[`maxLength - 11: valid 1`] = `""`;

exports[`maxLength - 12: valid 1`] = `""`;

exports[`maxLength - 13: invalid 1`] = `"maxLenVal must be at most 12 characters"`;

exports[`minLength - 11: invalid 1`] = `"minLenVal must be at least 12 characters"`;

exports[`minLength - 12: valid 1`] = `""`;

exports[`minLength - 13: valid 1`] = `""`;

exports[`moreThan - -5: valid 1`] = `""`;

exports[`moreThan - 0: invalid 1`] = `"number must be at more than 12"`;

exports[`moreThan - 25: valid 1`] = `""`;

exports[`mustBeOfType: invalid 1`] = `"mustBeofType must be a string"`;

exports[`mustBeOfType: valid 1`] = `""`;

exports[`mustMatchRegex: invalid 1`] = `"alphabet must be a, b, c, or d"`;

exports[`mustMatchRegex: valid 1`] = `""`;

exports[`noEmpty: invalid 1`] = `"string cannot be empty"`;

exports[`noEmpty: valid 1`] = `""`;

exports[`number: invalid 1`] = `"number must be a number"`;

exports[`number: valid 1`] = `""`;
